Governance 1.0 complete draft for discussion (#107)
operate-first/community: governance.md

---
title: Governance 1.0 DRAFT WIP
description: Governance 1.0 discussion draft
---
# Governance 1.0 `DRAFT WIP`

> This document contains a work in progress (WIP) draft of the initial governance of the Op1st project (Governance 1.0).

Reviewer: I would change Op1st to something like Operate First (Op1st) so that newcomers know what we're referring to

Reviewer 2: are we a project or a community? I feel we can have multiple projects but there's one overarching community:

`... governance of the Operate First community`

Author: > are we a project or a community? I feel we can have multiple projects but there's one overarching community:
> 
> `... governance of the Operate First community`

This is a good question, as there may not be a formal definition of these things in open source software, but they are generally used in common ways:

* **Community** is the group of people sharing a common interest and desire to work toward something in common. This usage of the word "community" is different from its dictionary definition or use in other fields, and is more like a shorthand for _[community of practice](https://github.com/theopensourceway/guidebook/blob/main/communities_of_practice.adoc)_ (CoP).
*  **Project** is the collective term, referring to the community of people and the formal structure the community creates and sustains in order to get their work done; a project is defined by a vision and driven by a mission, which are created and brought to life by the community.
* **Subproject** is typically a formal structure within a project, either organized at the top or within a SIG. This is the one that sounds most like what you describe above has "(having) multiple projects".

Organizations that contain multiple projects (as per the above usage) and are overarching are more like foundations, e.g. Python Software Foundation, Apache Software Foundation, and Cloud Native Computing Foundation.

So for example, if you are thinking of communities like OpenData Hub and Open Cluster Manager, they are forming into projects we are loosely affiliated with, but not even a loose confederation exists.You can have groups of projects that coordinate together without a formal structure.

Seeing all the +1s to your comment, what I'll do is add some clarifying language, and see if that suffices. Otherwise, we can start with "community" and do any changes as part of the 1.1 updates.

Author: > I would change Op1st to something like Operate First (Op1st) so that newcomers know what we're referring to

Thank you, yes, this was a very sloppy first draft paragraph. I will expand it; but as per #104 to use 'Op1st' for the community name, my proposal is that Op1st is no longer a shorthand for Operate First but rather the distinctly separate name for the Op1st community, which is implementing an Operate First envirionment.

```

Reviewer 2: Not sure I'm convinced about the `op1st` thing. But let's have that discussion in #104 :)

Author: ```suggestion
This document contains a work in progress (WIP) draft of the initial governance (Governance 1.0) of the community-formed project around the _[Operate First concept](https://openinfralabs.org/operate-first-manifesto/)_.
In this document, the community and project are referred to by the shorthand _Op1st_ (spoken as "op first").
```

The above language is intended to not-yet name the community ("Operate First" or "not-Operate First"). The explanation let's us use the shorthand throughout the document here, which is easy to change after we have a governance to enable us to discuss and #104 as a community, while not pre-deciding a community/project name.

Reviewer: This looks good to me!

> It also has a roadmap for additions coming in the next iteration of governance (Governane 1.1).

Reviewer: Governance, not governane

Author: Fixed, thanks.

> It also has a roadmap for additions coming in the next iteration of governance (Governane 1.1).

Author: ```suggestion
It also has a roadmap for additions coming in the next iteration of governance (Governance 1.1).
```

Author: Just in case this doesn't end up as planned because of cross-commits I made here, I'm reopening this discussion so it appears when looking after the merge is complete.

> It also has a roadmap for additions coming in the next iteration of governance (Governane 1.1).

Author: ```suggestion
It also has a roadmap for additions coming in the next iteration of the governance (Governane 1.1).
```

> It also has a roadmap for additions coming in the next iteration of governance (Governane 1.1).

Author: ```suggestion
It also has a roadmap for additions coming in the next iteration of governance (Governance 1.1).
```

> 

Author: > There could be a high level summary here, e.g.

## Summary

The community works in special interest groups (SIGs), who form around a common topic.

SIGs are operated by consensus and can self-organize.
A common organizational structure might be to appoint a chair and vice-chair by consensus, who organize and run meetings, and take charge of getting the group to consensus.

The overall project is operated by all SIGs in consensus.
The future roadmap includes the idea of committees, which are elected or selected bodies that operate outside of the SIG structure, and may direct the project above and beyond the SIGs.
For example, a future Technical Steering Committee might decide all overall technical directions, based on the advice and consent of the SIGs.

To start, we are initiating the Community SIG and the Operations SIG, all parts of the project fall under these SIGs.
The SIGs then self-organize and decide to sub-divide into further SIGs, and to form subprojects within one or more SIGs to actually do things.

The subprojects are where the work happens, the SIGs organize and guide their subprojects.

Reviewer 2: ...common organizational structure ~~might be~~ **is** to appoint a chair and vice-...

And ACK on the *Summary*. Just add it as an additional commit to the PR and we can comment on the section content

Reviewer 3: The community works in special interest groups (SIGs), that form around a common topic.

SIGs are operated by consensus and can self-organize.
A common organizational structure might be to appoint a chair and vice-chair by consensus who organize and run meetings and take charge of getting the group to consensus.

The overall project is operated by all SIGs in consensus.
The future roadmap includes the idea of committees, which are elected or selected bodies that operate outside of the SIG structure and may direct the project above and beyond the SIGs.
For example, a future Technical Steering Committee might decide all overall technical directions based on the advice and consent of the SIGs.

To start, we are initiating the Community SIG and the Operations SIG. All parts of the project fall under these SIGs.
The SIGs then self-organize and decide to sub-divide into further SIGs and to form subprojects within one or more SIGs to actually do things.

The subprojects are where the work happens, and the SIGs organize and guide their subprojects.

Reviewer 4: The summary should perhaps start with (a tl;dr of) the why and not with the how

Author: > ...common organizational structure ~might be~ **is** to appoint a chair and vice-...

I get your intent here, I made that change but also expanded the paragraph just a bit to clarify the "one common structure" is not a dictate.

The key is to let SIGs self-organize rather than restrict them to start, and only restrict structures that prove problematic (rather than are predicted to be problematic.)

> And ACK on the _Summary_. Just add it as an additional commit to the PR and we can comment on the section content

And also AIUI, we can provide fixes as comments, which can then be merged like they were a direct commit.

Author: Updated in d1adfd2

> 

Reviewer 5: What is the process for SIG chair/vice chair elections?

Reviewer 2: yes, should be defined in the SIG creation process- but not here in the overview section.
and maybe we can postpone this detail for v1.1

Author: > yes, should be defined in the SIG creation process- but not here in the overview section. and maybe we can postpone this detail for v1.1

We must postpone this kind of detail for a few reasons:

* The process for choosing leaders is an important decision the community needs to make collectively, but we can't do that until we have ratified and are using a process for making decisions as a community. Governance 1.0 is focused on enabling us to make decisions, and tries to make as few decisions on behalf of the community in the 1.0.
* We want to preserve the capability for SIGs to self-organize in different ways, at least until we have a way for the community to decide why and how to restrict that.

We can definitely put in some kind of example/try this language so the initial SIGs can kickstart and refine.

## Op1st Community Governance 1.0 (MVG)

A healthy open source project is one that demonstrates open practices, uses open infrastructure, and cultivates an open culture with the goal of becoming more sustainable.

### Principles

> As a new cloud-native community growing up at the footsteps of massively scaled and successful projects such as Kubernetes and OpenInfra, we want to be careful to learn from those who came before us, while forging our own path into the challenges we've chosen to undertake.

Reviewer 2: ~~cloud-native~~ though this is where we currently are, this might sound non-inclusive to individuals

Author: ```suggestion
As a new community with a vision to grow, we are following in the footsteps of many massively scaled and successful projects. We want to take care to learn from those who came before us, while forging our own path into the challenges we've chosen to undertake.
```

> As a new cloud-native community growing up at the footsteps of massively scaled and successful projects such as Kubernetes and OpenInfra, we want to be careful to learn from those who came before us, while forging our own path into the challenges we've chosen to undertake.

Reviewer 3: those who came before us while forging our own path into the challenges

Author: How does my variation sound?

(BTW, I just learned you an click the +- icon above the comment box and it will let you make a suggestion for changs, which can then be committed from here.)

Author: Resolving for now as no further objection was raised. If the language needs adjusting, we can work on it as part of Gov 1.1.

To stand up this Governance for a 1.0 good-enough minimal-viable-governance (MVG), we are looking to our key predecessors for some of our starting points.
As an initial set of principles we are combining elements of the [Kubernetes community principles](https://github.com/kubernetes/community/blob/master/governance.md#principles) and the [OpenInfra Foundations Four Opens](https://openinfra.dev/four-opens/).

#### Op1st Community Principles 1.0
Our mission requires us to use open source principles in ways beyond licensing the code we write, but at the core we are an open source community and want to adhere to the best possible principles for creating an open, welcoming, and inclusive community.

These are the Op1st community principles we adhere to:

> 1. **Open Source** : Any content we create is released under an open source license. Content includes software, documentations, configuration files, and training materials.

Reviewer: Should we specify/determine the license here itself?

Reviewer 2: 👎 the license should not be enforced.

Reviewer 2: @quaid please add `operational data` to the list of content. Or just `data`

Reviewer 6: Similar to @durandom 's comment, since this is about operate first, should we add add "operational code" or "code necessary to operate software".

Reviewer 4: Content could be documentation. Should the sentence then say 'open source or open content license'?

Author: ```suggestion
1. **Open Source** : Any content we create is released under an open work license. Content includes all software, documentations, configuration files, data, and training materials.
"All software" includes the code necessary to operate software, AKA _operational code_. 
```

Author: @durandom @stefwalter I think this covers your cases?

@pilhuhn very good point, and because we have other output such as data, AI/ML artifacts, and artwork (logos, diagrams, etc.), I changed to the completely-inclusive "open work" such as how [Open@RIT uses it](https://drive.google.com/file/d/1DjjuEnfJKxfDFdktDoc_3Xx2ne8xp9fX/view).

Reviewer 2: yes, full ACK. This looks good

Reviewer 6: @quaid Works for me. Although I would have liked to see operational code clearly called out, since this is often ignored, I understand that it's covered by the principle.

Author: > @quaid Works for me. Although I would have liked to see operational code clearly called out, since this is often ignored, I understand that it's covered by the principle.

You know what, you've convinced me—when working on mindset changes, it helps to highlight these definitions that might not be assumptive to everyone yet. I'll work this in a way that benefits the cause.

> 1. **Open Source** : Any content we create is released under an open source license. Content includes software, documentations, configuration files, and training materials.

Reviewer 6: Lets link the words [open source license] to the OSI approved license list: https://opensource.org/licenses

Reviewer 6: One more factor. People will ask (as they have already asked me) why the AGPL was not mandated for some/all of this content. Do we have an ADR on this? Does that decision need to be recorded? Otherwise, this will be an obvious FAQ, and likely sore point.

Author: > Lets link the words [open source license] to the OSI approved license list: https://opensource.org/licenses

Surprisingly, this is why I didn't want to link to a specific list of licenses. We should have a community discussion around this. For example, I recommend using the [Fedora licensing page](https://fedoraproject.org/wiki/Licensing:Main?rd=Licensing) with some contextual explanations, such as how to use the list. Some people will want a specific subset of licenses, and those arguments should be heard.

If we start by linking to the full list of licenses, even if it's just to refer to what we mean by "open source licenses", we have defined a list for people. Then if we want to do any of the above suggestions as a community, we are in the position of taking something away from people, after just giving it to them.

So the reasoning here is to leave it undefined but clear enough, then clarify that in Governance 1.1.

(We'll come back to this PR to mine it for 1.1 specifics, which I'll mark with a :heart: for now.)

How does that sound to you?

Reviewer 6: That works for me. As long as we describe this thought process, otherwise the question will arise in people's minds.

Author: Resolving this to make it clear there is no further action needed for Gov 1.0. We'll come back to this to seed the discussions for Gov 1.1.

> 1. **Open Source** : Any content we create is released under an open source license. Content includes software, documentations, configuration files, and training materials.

Reviewer 7: Nit: Do we need to define what "content" is here? If so, we should be more inclusive I think:

```suggestion
1. **Open Source** : Any content we create is released under an open source license. Content includes software, documentations, configuration files, deployment manifests, telemetry data, training materials, and any other content meaningful to the project.
```

Author: Love your use of the Suggestion feature! I just learned about that today.

I did a suggestion [above](https://github.com/operate-first/community/pull/107/files/092e770de008cf2c8ccfd95f528b3326618d2baa#r771018629), and I think it works to cover your use cases -- configuration should cover deployment manifests (and if not then we can keep it), and data covers all data.

I also think I'm going to change my fix to "includes but is not limited to" as a different way to include things not listed.

Reviewer 7: I like that :+1: :slightly_smiling_face:

  - This is crucial to fulfilling our mission around open sourcing operations

> 2. **Welcoming and respectful** : We intend to be an open, welcoming, and inclusive community.

Author: Thanks for thinking of this and suggesting it. I'm going to provide all my context here so everyone can understand, add to, and share the reasoning.

I pulled the three-item turn of phrase together out of a sort-of modern sentiment—it's similar to wording used in projects perceived as inclusive. The general approach is to focus on [being as inclusive as possible when designing an architecture for making a more diverse and equitable community](https://www.theopensourceway.org/the_open_source_way-guidebook-2.0.html#_to_build_diverse_open_source_communities_make_them_inclusive_first). When people come to the community and find it accessible, regardless of who they are, they can more easily and willingly bring their voices to the common table.

But recently I've been thinking about how to bring _equitability_ into these processes, statements, and outcomes, especially since the [All In Open Source](https://allinopensource.org) project started with the goal of open sourcing diversity, equity, and inclusion. I figure one of the first steps is to talk about equity, and I appreciate you seeing it here as an ideal we put forth. That means it's something we talk about in the SIGs—how to be more equitable, how to measure our success at that, how to improve—and thus something we can improve on over time as the project grows more diverse.

```suggestion
2. **Welcoming and respectful** : We intend to be an open, welcoming, equitable, and inclusive community.
```

> 2. **Welcoming and respectful** : We intend to be an open, welcoming, and inclusive community.

Author: @akrawczy do you have any preferences of these variations?

Diversity is our strength.
We have and follow a Code of Conduct.

>   - We currently follow the OpenInfra Foundation Code of Conduct as a member of the OpenInfra Labs.

Reviewer: Should link to this

Author: > Isn't this redundant with the later "Code of Conduct" section? https://github.com/operate-first/community/pull/107/files#diff-fd3edf48a550fc1e06e551c9de0928dc1df51b9aef61139df0bed27ed2d9b468R38

No, I don't see it as redundant or repetitive. This reference to the Code of Conduct is in a section stating the overarching community principles, one of which is to have and follow a code of conduct.

The other link is the actual section where the governance talks about the specific project Code of Conduct, and acts as the in-document referenced location for any other reference in the document.

(I believe it would even be proper and reasonable to include the entire text to a Code of Conduct within the Governance itself, as part of having an entirely self-referential governance. But I'll save the rest of that discussion for when we get to actually deciding as a community what our Code of Conduct should say and do.)

Author: Done, thanks, linking the first one to our own URL as the canonical URL for our CoC. That way, if we change the Code of Conduct in any way, the URL for it still works, compared to if we linked directly the OpenInfra Foundation CoC.

Then the full title of that CoC balances and makes sense with the canonical link to our own domain.

>   - We currently follow the OpenInfra Foundation Code of Conduct as a member of the OpenInfra Labs.

Reviewer 7: Isn't this redundant with the later "Code of Conduct" section? https://github.com/operate-first/community/pull/107/files#diff-fd3edf48a550fc1e06e551c9de0928dc1df51b9aef61139df0bed27ed2d9b468R38

Author: > Isn't this redundant with the later "Code of Conduct" section? https://github.com/operate-first/community/pull/107/files#diff-fd3edf48a550fc1e06e551c9de0928dc1df51b9aef61139df0bed27ed2d9b468R38

No, I don't see it as redundant or repetitive. This reference to the Code of Conduct is in a section stating the overarching community principles, one of which is to have and follow a code of conduct.

The other link is the actual section where the governance talks about the specific project Code of Conduct, and acts as the in-document referenced location for any other reference in the document.

(I believe it would even be proper and reasonable to include the entire text to a Code of Conduct within the Governance itself, as part of having an entirely self-referential governance. But I'll save the rest of that discussion for when we get to actually deciding as a community what our Code of Conduct should say and do.)

> We have and follow a Code of Conduct.
>   - We currently follow the OpenInfra Foundation Code of Conduct as a member of the OpenInfra Labs.

Author: ```suggestion
We have and follow a [Code of Conduct](https://operate-first.cloud/community/code_of_conduct/).
  - We currently follow the [OpenInfra Foundation Code of Conduct](https://www.openstack.org/legal/community-code-of-conduct/) as a member of the OpenInfra Labs.
```

> 3. **Open Design** : We follow a transparent and open process for planning and designing any content, services, and features of Op1st.

Reviewer: Should we mention/link to the ADR process here?

Reviewer 2: ADR is one implementation of this process, and described later in this document.

Ideas and contributions are accepted according to their technical merit and alignment with project objectives, scope, and design principles.
4. **Open Work** : Two of our core tools for successful community building are transparency and accessibility.
To support that, work and collaboration must be done in public to the greatest degree possible.

> Refer to the section on Community Groups and SIGs in particular for details on how this is performed.

Reviewer 2: You can link to the sections: 
https://docs.github.com/en/github/writing-on-github/getting-started-with-writing-and-formatting-on-github/basic-writing-and-formatting-syntax#section-links

Author: Thanks, I think this is the correct syntax (the links are accurate, at least currently in the MD file in my fork rendered by GitHub ...)

```suggestion
Refer to the section on [Community Groups](#community-groups) and [SIGs](#sigs) in particular for details on how this is performed.
```

> 5. **Truly Open Community** : The best way to build a sustainable open source community is to ensure the community is a cohesive, inclusive, and equitable place where all the voices are heard and anyone can rise to leadership positions.

Reviewer 3: Is this redundant to #2? Is there a way to differentiate it?

Author: > Is this redundant to the second point? Is there a way to differentiate it?

I do see the differentiating, but it's not in the wording; and I think it needs some focus for the last one.

How does this jibe for you?

```suggestion
5. **Sustainable Open Community** : The best way to build a sustainable open source community is to ensure the community cohesion is through openness and transparency, ensuring all the voices are heard and anyone can rise to leadership positions.
The more diversity of voices at the decision table, the better our ability to improve conditions such as inclusion and equity.
Sustainable comes through continuous improvement of those areas, such as through centering the agenda of those people most marginalized and at risk in each given context (which results in things being more accessible for everyone.)
```
I see another challenge is where it says effectively in two places "open ... community". I'm hoping this helps with that.

### Code of Conduct

> We are using the [upstream OpenInfra Foundation Code of Conduct](https://openinfra.dev/legal/code-of-conduct) until we can convene the new Community SIG to lead an open effort to create our own, modern code of conduct.

Reviewer 2: ~~~modern~~~ this implies theirs is old and outdated, which I hope it's not 😄

Reviewer 8: Definitely not "outdated" as the board of directors just voted to approve updates to it a few months ago.

More to the point, the idea is for that Community Code of Conduct to be usable by all projects represented by the Open Infrastructure Foundation. If you have feedback on how to improve it to be more useful for Operate First, then that would be entirely welcome. Probably the most fruitful avenues for amending it would be to start a discussion on the foundation mailing list, or join the Diversity and Inclusion Working Group meetings as they have a fair amount of experience with the current state of the document and process of engaging with the foundation's staff, board, and legal counsel in order to drive improvements there or in similar policy documents.

Reviewer 7: I don't think the "until" clause should be listed here. Is there a reason to say that "things may change in future" in here? I think that's actually implied in all sections, isn't it?

```suggestion
We are using the [upstream OpenInfra Foundation Code of Conduct](https://openinfra.dev/legal/code-of-conduct).
```

Reviewer 4: Should there be a mention that individual projects that are deployed on Op1st (e.g. Strimzi / Kafka) may have their own CoC, which would apply appropriately?

Author: @durandom @fungi @tumido You raise good questions, let me break down what is happening in this sentence; we can come to conclusion on what it should be as a good-enough to start.

1. This community never chose<sup>[1](#fn01)</sup> its code of conduct, because this community has not previously had a method for making a community-wide decision in an open source way. One goal of this section is to make it clear that we need to have the actual step of this community discussing and deciding on a code of conduct. It may be as simple as agreeing to continue using the OI Foundation Code of Conduct.
2. _However_, there is activity right now in the ecosystem looking at the process of adopting codes of conduct, the templates they are modeled from, and how projects administer and enforce them. We would be remiss to skip the opportunity to look at things with the freshest perspectives available.
3. Finally, to @pilhuhn's point, we need to thing of our CoC in light of also being a platform for the success of other projects, and we need to consider that through so we remain aligned.

@tumido I agree fully about all things being up for change in the 1.1 version. My reason for saying this in particular among all others is for those who are looking beyond the existence of a Code of Conduct for a community actively practicing inclusion. Currently, that means demonstrating an awareness of the above points and where we are at this moment in history with regards to practices and discussion around codes of conduct. Which basically is, "Currently our practices are under discussion, it's worth taking the extra time to talk it through especially for a new community."

@fungi thank you, those offers are much appreciated and give us some good ways to setup a transparent and equitable process for this community. The Community SIG will drive this discussion (and I specifically invite you to participate in that SIG, at least if you are able to for this first year as we work out many initialisms), the discussion might look like this:

1. We put out a call for participation in a CoC working group, giving a few weeks to form and have the first meeting so we can get the most participation.
    - We MUST have an equitable asynchronous process for participation, including fairly long time arcs, regular reminders of status, zero requirement for real-time meetings (or perhaps requiring zero real-time meetings?), and so forth.
2. Once the CoC WG formally engages, some level of working together and bonding is necessary to be able to engage with the OpenInfra Foundation processes.
3. When ready, the WG engages with the OI Foundation D&I WG to begin a working relationship and see where things are at that point.
    - In my mind, it would be great if we could surface the majority of discussion points and change requests from the community via these first few interactions. Then anything in the future has a change management process it can follow and typically a longer lead time.
4. If any changes or other items are needed, the two WGs collaborate on a presentation or proposal to the OI Foundation Board.

How about these adjustments?
```suggestion
We are using the [upstream OpenInfra Foundation Code of Conduct](https://openinfra.dev/legal/code-of-conduct)(CoC).
In creating the next Governance 1.1 we will have an actual process to make decisions as a community. We can then either formally ratify this same CoC, or the Community SIG can convene a cross-project working group to discuss and decide an inclusive and equitable process for a new code of conduct.
```

<a name="fn01">1</a>: I couldn't find a CoC when I did my initial audit of this community. It not only wasn't linked anywhere, it wasn't clear there even was one. That's a Sev1 (high priority thing to fix) from a community architect point of view, so I paused my audit to file my first PR for this project: https://github.com/operate-first/operate-first.github.io/pull/244 from 9c2d0e6.

Reviewer 9: I went looking on both the OpenInfra Labs and Operate First websites and they both point to the OIF CoC. I'm not sure if you added the link on the Operate First site after this document was created @quaid or if it was there already there though. @fungi might know the process more of how projects are set up within OIF as to whether utilizing the OIF CoC is part of the agreement. 

As far as working with the OIF D&I WG, we are always inviting all the projects to the meetings, I'm actually not sure if any of the sub-projects have their own D&I groups or not. The WG is available to all projects to help with D&I issues from helping with Inclusive Naming, D&I Surveys, etc. and reports back to the Foundation Board. While originally it was just OpenStack as the only project at the time, it never was actually part of OpenStack if that helps clarify anything.

Reviewer 8: Amy's comment just reminded me... when I replied by E-mail a few weeks ago, GitHub mis-threaded it into general comments instead of into the file comments, so I'm copying it here for better context. Sorry!

On 2021-12-18 16:05:13 -0800 (-0800), Karsten Wade wrote:
[...]
> 1. This community never chose<sup>[1](#fn01)</sup> its code of
> conduct, because this community has not previously had a method
> for making a community-wide decision in an open source way. One
> goal of this section is to make it clear that we need to have the
> actual step of this community discussing and deciding on a code of
> conduct. It may be as simple as agreeing to continue using the OI
> Foundation Code of Conduct.

I suppose it's all a matter of perspective. The Open Infrastructure
community chose a code of conduct and continues to maintain it with
the help of its foundation, members, and board of directors. The
OpenInfra Labs project chose to adopt that code of conduct and help
shape any future revisions thereof by joining the foundation which
represents the Open Infrastructure community. Op1st, as a subproject
of the OpenInfra Labs project (or "key development activity" in the
words of the project's website), is covered by that project's
commitment to follow the Open Infrastructure Community Code of
Conduct.

> 2. _However_, there is activity right now in the ecosystem looking
> at the process of adopting codes of conduct, the templates they
> are modeled from, and how projects administer and enforce them. We
> would be remiss to skip the opportunity to look at things with the
> freshest perspectives available.
[...]

I completely concur, this is an excellent opportunity to revisit the
intent and wording of the Community Code of Conduct and help revise
it to the benefit of the Open Infrastructure Community, of which
Op1st is a part.

> @fungi thank you, those offers are much appreciated and give us
> some good ways to setup a transparent and equitable process for
> this community. The Community SIG will drive this discussion (and
> I specifically invite you to participate in that SIG, at least if
> you are able to for this first year as we work out many
> initialisms), the discussion might look like this:
[...snip proposed plan...]

This sounds like a reasonable approach to me, and I'm happy to
(continue to) join in these discussions. Just to underscore my
perspective, Op1st does already have a code of conduct which it
shares with a community of projects. Much like open source software,
open communities are easy to fork but such forks should be taken
with care, and ideally start with attempts to collaborate in the
maintenance of the resources shared with the rest of the community.

> <a name="fn01">1</a>: I couldn't find a CoC when I did my initial
> audit of this community. It not only wasn't linked anywhere, it
> wasn't clear there even was one. That's a Sev1 (high priority
> thing to fix) from a community architect point of view, so I
> paused my audit to file my first PR for this project:
> https://github.com/operate-first/operate-first.github.io/pull/244
> from 9c2d0e6.

To be clear, the CoC has been declared at the bottom of each page of
the project Web site since
https://github.com/open-infrastructure-labs/openinfralabs-website/commit/78dd7e1
merged in August of 2020. I agree it's not a bad idea for
subprojects to mention the CoC in their documentation as well, but
there seems to be a lot of fragmentation across OpenInfra Labs at
present, pointing to a need to establish a clear project governance
while the project itself works toward increased cohesion (made more
evident by one of its subprojects drafting its own formal governance
even before the project has initiated such a discussion).

Author: > I went looking on both the OpenInfra Labs and Operate First websites and they both point to the OIF CoC. I'm not sure if you added the link on the Operate First site after this document was created @quaid or if it was there already there though. @fungi might know the process more of how projects are set up within OIF as to whether utilizing the OIF CoC is part of the agreement.

Other way around, I had us add the OIF CoC in May 2021. At that time no one in the project was aware of any process or for that matter seemed to consider themselves to be covered by any formal code of conduct. This was in operate-first/operate-first.github.io@9c2d0e6 .
 
> As far as working with the OIF D&I WG, we are always inviting all the projects to the meetings, I'm actually not sure if any of the sub-projects have their own D&I groups or not. The WG is available to all projects to help with D&I issues from helping with Inclusive Naming, D&I Surveys, etc. and reports back to the Foundation Board. While originally it was just OpenStack as the only project at the time, it never was actually part of OpenStack if that helps clarify anything.

Thanks, that helps. Now that we have are going to have a governance, we can actually have a way to interact with the OIF, including on DEI matters.

Author: > I suppose it's all a matter of perspective. The Open Infrastructure community chose a code of conduct and continues to maintain it with the help of its foundation, members, and board of directors. The OpenInfra Labs project chose to adopt that code of conduct and help shape any future revisions thereof by joining the foundation which represents the Open Infrastructure community. Op1st, as a subproject of the OpenInfra Labs project (or "key development activity" in the words of the project's website), is covered by that project's commitment to follow the Open Infrastructure Community Code of Conduct.

That is the logic I followed in opening operate-first/operate-first.github.io#244—I wouldn't have pushed an active community to accept being under a code of conduct without any kind of process to get there, but it was also reasonable to conclude that by dint of relationship we were already under the OIF CoC.

It has been since May 2021 that the conversations about CoCs have grown, so that was part of the reasoning here. The other part is that no matter what, it's a good step for the Op1st community to formally recognize their CoC. E.g. as part of the discussion we'll want to have some training materials, resources, reporting pathways, etc. so community members have a stronger understanding.
 
> To be clear, the CoC has been declared at the bottom of each page of the project Web site since [open-infrastructure-labs/openinfralabs-website@78dd7e1](https://github.com/open-infrastructure-labs/openinfralabs-website/commit/78dd7e1) merged in August of 2020. I agree it's not a bad idea for subprojects to mention the CoC in their documentation as well, but there seems to be a lot of fragmentation across OpenInfra Labs at present, pointing to a need to establish a clear project governance while the project itself works toward increased cohesion (made more evident by one of its subprojects drafting its own formal governance even before the project has initiated such a discussion).

On the former point, I think it's crucial and it's recommended practice to better highlight Codes of Conduct. In fact, I'm not happy with where it is right now and have Community SIG issue operate-first/community#54 about that.

On the latter point, I honestly considered going to OI Labs first to get governance there first so we could be given the charter to create an Op1st governance. I think wisely I'm keeping within where I can help make things happen, but I would he happy and interested in helping OI Labs put a governance together.

(I'm always finding it important in conversations for people to understand the relationship between the conceptual work happening in OI Labs and the prototype work happening in Op1st. Getting governance pieces to fit as parts of the puzzle will help, especially if it's a nice diagram on the OI Labs site.)

Author: OK, the way I worded it in the suggestion above made it into one of these commits :) and it seems we're in agreement to proceed and keep discussing these topics in their new home of the Community SIG.

I'm resolving this to make it clear there no more actions needed for this PR, and we can come later for seeding future Gov 1.1 discussions.

### Community membership
**Note:** This document is a work in progress, especially these membership guidelines

We are opening the Governance 1.0 directly borrowing from the [Kubernetes community group definitions](https://github.com/kubernetes/community/blob/master/community-membership.md), then reviewing and refining for the Governance 1.1.

This doc outlines the various responsibilities of contributor roles in Op1st.

> The Op1st project is subdivided into subprojects under SIGs.

Reviewer 2: s/project/community/

Reviewer 2: The Operate First community is organized primarily into Special Interest Groups, or SIGs. Specific work efforts within SIGs are divided into subprojects. 

https://github.com/kubernetes/community/blob/master/governance.md#sigs
https://github.com/kubernetes/community/blob/master/governance.md#subprojects

Author: Discussed and resolved in [this comment above](https://github.com/operate-first/community/pull/107/files/092e770de008cf2c8ccfd95f528b3326618d2baa#r770799637); if there is disagreement or clarification, let's keep it in that thread.

> Responsibilities for most roles are scoped to these subprojects.

Reviewer 2: Maybe link https://stumblingabout.com/2011/01/27/ownership-roles-and-responsibilities/ 
This is also linked from the k8s governance files. It explains the interaction of those roles quite well

Author: I totally agree with the concept, but I always hesitate to link to external, potentially ephemeral URLs. That is, if we rely upon that link for the understanding without having our own local, refined version, we are rudderless without that person's domain being up and running. (I know, I know, let's not talk about github.com, quay.io, etc. etc.)

As an interim good-enough, I'll make a change that identifies it as a separate reference and we can put it on the list to update for 1.1. (One way may to look for the gaps in the [Open Source Way content](https://www.theopensourceway.org/the_open_source_way-guidebook-2.0.html#_community_roles) and that link above, then fix the OSW guidebook with content to fill the gaps.)

Author: ```suggestion
Responsibilities for most roles are scoped to these subprojects.
"[Ownership, roles and responsibilities](https://stumblingabout.com/2011/01/27/ownership-roles-and-responsibilities/)" is a good external reference for understanding how the project roles interact.
```

| Role | Responsibilities | Requirements | Defined by |
| -----| ---------------- | ------------ | -------|

> | Member | Active contributor in the community | Sponsored by 2 reviewers and multiple contributions to the project | Op1st GitHub org member|

Reviewer 10: What is a "contribution"? This is mentioned below by Misc, and can be defined here to be more than *just* code contributions.

Also, "Active" needs clarification. Does someone stop being a member when they're no longer active? If so, what does active mean?

Author: Thanks @rbowen @mscherer @schwesig for these comments and observations. I've made this commit: 279810a to address these issues

One thing I'm curious about is how much our assumptions come into play. For example, although "contributions tracked in GitHub" doesn't actually say "only code or docs really matters", I do agree others will read and draw the that or similar conclusion you are. Makes sense to be more clear anyway.

But it does show a bias of old practices. For example, @mscherer's suggestion about 'we focus on ... git based" for now, will extend in the future presumes we won't be having every single role and contribution interact with a git workflow.

By the contrary, I can imagine the vast majority (>95%) of contributions can be tracked back to directly or implied/inferred by git workflow activity. Take event planning as an example: all of that planning, staffing schedules, messaging indexes, etc. are going to be appearing in the form of issues, PRs, comments, reviews, meeting notes and recordings, and so forth.

| Reviewer | Review contributions from other members | History of review and authorship in a subproject | [OWNERS] file reviewer entry |

> | Approver | Contributions acceptance approval| Highly experienced active reviewer and contributor to a subproject | [OWNERS] file approver entry|

Reviewer 11: Having a special role based on review and approver seems to mean contributions will mostly be around code (or things that could be expressed in the same terms, like docs stored in git, etc). This implicitly send the signal that non-code-like contributions would be less important, as this do not seems to appear in the governance document. 

Could some notes on that be explicitly added ? 

EG, that we focus for now on that type of contribution (eg, git based), but we will be working to extend the governance for others types of contributions based on the activities, or something in that spirit ?

Reviewer 12: I agree with your concern. But I don't have a better solution, yet. Solution in what (content) and how (technical solution)
But maybe this could/should be a topic for v1.1 or later?

Reviewer 10: Reviewing contributions is not limited to code contributions, but maybe this could be called out explicitly?

Author: Addressed in 279810a

| Subproject owner | Set direction and priorities for a subproject | Demonstrated responsibility and excellent technical judgement for the subproject | [sigs.yaml] subproject [OWNERS] file *owners* entry |

### Community groups
The Op1st community is initiating Governance 1.0 with only one type of community group, Special Interest Groups (SIGs).
The purpose of doing so is to empower SIGs to gather interested people around a subject area, then decide and implement that into governance.
For example, one of the first acts of the Community SIG is to establish a milestone schedule for defining and holding elections for future steering committee(s).
Governance 1.1 will therefore include details on new community groups (subprojects and working groups).

#### SIGs
Special Interest Groups (SIGs) are persistent open groups that focus on a part of the project. SIGs must have open and transparent proceedings.

> Anyone is welcome to participate and contribute provided they follow the Op1st Code of Conduct.

Reviewer 2: link to the CoC

Author: ```suggestion
Anyone is welcome to participate and contribute provided they follow the Op1st [Code of Conduct](https://www.operate-first.cloud/code_of_conduct).
```

> Anyone is welcome to participate and contribute provided they follow the Op1st Code of Conduct.

Reviewer 6: Is any member able to open a SIG? When we say "Anyone" here ... what does that actually mean WRT a SIG?

Author: > Is any member able to open a SIG? When we say "Anyone" here ... what does that actually mean WRT a SIG?

I've been mulling on this one, as I wasn't sure if a clarification of meaning was needed, or have you hit upon something that needs a group consensus. I think it's both. :)

1. IMO, any community member (as defined in that section) should be empowered to propose a SIG, according to whatever criteria the Community SIG defines (X type of mission, Y number of members, etc.) But I don't think it should be easy or encouraged for people to join the project with the singular goal of opening a new SIG (unless invited by the community to do so.) That is, I see SIGs arising from community-identified needs, so an outsider would need to sync with the community enough to "be one of us" in order to create consensus around the need.
2. I do think it's OK and encouraged to join the project specifically to join a SIG. In that case, the language here is intended to remind people that participating in a SIG is a way of participating in the project overall regardless of interest and focus. Even a one-off contribution process should have the clear expectation that a one-off contributor is following the Code of Conduct. So the language here is to make it clear that one is welcome to drop into a SIG, offer a patch, and disappear forever ... but it all has to be done under the same aegis as a regular, repeat contributor

Reviewer 6: The work in https://github.com/operate-first/community/commit/d1adfd2e8848392404e1a9052e9e8156dbfdc1a8 covers the SIG lifecycle. I believe this provides enough of an answer to this question. The remainder can be decided by the community and not set in stone here.

The purpose of a SIG is to own and develop a set of subprojects.

> A SIG can have its own policy for contribution, described in a README or CONTRIBUTING file in the SIG folder in this repo (e.g. sig-community/CONTRIBUTING.md), and its own mailing list, chat channel, etc.

Reviewer 6: Is there any part of the project that is not in a SIG? If so, what differentiates the community work from SIG work? Lets answer this question and related stuff here.

Author: > Is there any part of the project that is not in a SIG? If so, what differentiates the community work from SIG work? Lets answer this question and related stuff here.

This seems like it's answered in the 1.1 future groups that follow? In that, the idea there is work outside of a SIG and its subprojects is addressed (committees, working groups) is included without yet prescribing the details and points of difference.

My reason for leaving the answer to your question until the 1.1 work is these are the kinds of things that seem writ in stone if they come in the first governance—instead of being ideas needing generating, it becomes existing policy that needs to be debated and changed. Yet the original debate to create the policy hasn't taken place! So then we're going back to interpret past intentions without the discussion record, etc.

What I think we can do best is capture all of these very good discussion points as starting points for the Community SIG work on Governance 1.1.

Reviewer 6: Makes sense. The description below in 1.1 covers this.

> 

Author: > MISSING SUB-SECTION - I forgot to include this, it should read like this:

#### Initiating SIGs

In order to get things started in a way to encourage transparency and participation, we are going to only initiate two SIGs to start, the Community SIG (`sig-community`) and the Operations SIG (`sig-ops`).
These two SIGs are immediately empowered through this governance to create new SIGs and/or subprojects.

To create subprojects, a SIG only needs to have consensus within itself.

To create a new SIG:

1. A SIG recognizes there is a need for a new SIG.
2. For example, the Community SIG decides to create the Data Science SIG right away, to formalize the already existing activity there.
3. Once the individual SIG is in consensus, the proposal is brought to all SIGs.
4. With all SIGs in consensus, the new SIG is formally created.

A similar process is followed when the Community SIG is ready to present recommendations for Governance 1.1 that creates committees.

Reviewer 2: ACK. Add as a commit to the PR so we can comment 😄

Reviewer 11: So that's how SIG are created, but what about SIG being deprecated ? EG, how would a SIG decide it is no longer important, if there is no longer anyone in it, etc, etc.

Author: > ACK. Add as a commit to the PR so we can comment smile

Done in d1adfd2 .

They are good-enough placeholders for discussions.

* **Committees** are named sets of people that are chartered to take on sensitive topics. This group is encouraged to be as open as possible while achieving its mission but, because of the nature of the topics discussed, private communications are allowed. Examples of committees include the Technical Steering Committee or an overarching Steering Committee, and things like security or code of conduct.
* **Subprojects** Each SIG can have a set of subprojects. These are smaller groups that can work independently. Some subprojects will be part of the main Kubernetes deliverables while others will be more speculative and live in the kubernetes-sigs github org.
* **Working Groups** are temporary groups that are formed to address issues that cross SIG boundaries. Working groups do not own any code or other long term artifacts. Working groups can report back and act through involved SIGs.
* **User Groups** are groups for facilitating communication and discovery of information related to topics that have long term relevance to large groups of Kubernetes users. They do not have ownership of parts of the Kubernetes code base.

> See the full governance doc for more details on these groups.

Reviewer 10: Is this level of subdivision actually warranted? Are you creating complexity before it is needed?

Author: Heh heh, especially when it's just a copy and paste still of the Kubernetes docs. I'll fix these in the next comment.

Fortunately, these are not being created or proposed at this point; this list is in a section labeled as `1.1` and `future`, and are placeholders for discussion. I'll add to the intro paragraph make it clear the discussion is intended to happen after Governance 1.0 is in use, and the Community SIG can begin the actual discussions.

Author: My fixes are in 48d2725:

> #### Governance 1.1 potential future groups
> [These descriptions](https://github.com/kubernetes/community#governance) are borrowed from [Kubernetes governance around community groups](https://github.com/kubernetes/community/blob/master/governance.md#community-groups).
> They are only placeholders for discussions, lightly edited.
> 
> **Subprojects** Each SIG can (or may? or should?) have a set of subprojects. These are smaller groups that can work independently. Some subprojects will be part of the main project Op1st deliverables while others will be more speculative.
> **Working Groups** are temporary groups that are formed to address issues that cross SIG boundaries. Working groups do not own any code or other long term artifacts. Working groups can report back and act through involved SIGs.
> **User Groups** are groups for facilitating communication and discovery of information related to topics that have long term relevance to large groups of Op1st users. They do not have ownership of parts of the Op1st code base, but they may contribute to other types of content and open works.

Author: Readable in context in my fork:  https://github.com/quaid/community/blob/main/governance.md#governance-11-potential-future-groups

### Decision making

While the project is in the early stages, we are favoring a balance of including all voices and keeping a rapid pace of progress.
To accomplish this balance, we are using a _consensus decision model_ that gives time and space for everyone to voice concerns with the proposal.

There are two types of consensus models we use, the lightweight and the full:

* **Lightweight consensus** occurs when a decision does not have wide or sweeping effects and is considered approved if no one raises any blocking objections after several days.

> In practice, an email is sent to a SIG or main community list, and if no one raises an objection after three days, the proposal passes.

Reviewer 6: Does someone need to note it's a proposal? Or transparently note that it has passed?

Reviewer 4: s/3 days/3 work days/

Author: ```suggestion
In practice, a proposal email with sufficient detail is sent to a SIG or main community list, and if no one raises an objection after three working days (i.e., Saturdays and Sundays are not counted), the proposal passes.
```

> * **Full consensus** decisions require 3 yes votes (+1) and no objections (-1’s) and votes should be left open for at least 72 hours.

Reviewer 12: Misread the "at least".
+1 to "at least 72 hours"

> * **Full consensus** decisions require 3 yes votes (+1) and no objections (-1’s) and votes should be left open for at least 72 hours.

Reviewer 4: weekend is problematic. We don't want to force folks to watch op1st on weekends to make sure they don't miss anything.

> * **Full consensus** decisions require 3 yes votes (+1) and no objections (-1’s) and votes should be left open for at least 72 hours.

Author: Thanks all, I agree these are important clarifications to make. Do these changes seem more fair and equitable?

```suggestion
* **Full consensus** decisions require 3 yes votes (+1) and no objections (-1’s) and votes should be left open for at least 72 hours (not counting weekend days) to ensure all voices are heard.
72 hours is a minimum and should be increased for decisions with a likely broad impact, that involve contentious issues, or for any other reason that increases fairness in reaching consensus.
Specifically, these rules must **not** be used to intentionally or accidentally disenfranchise contributors.
For example, if a 72 hour window goes over a weekend and means a significant decision closes on a Monday, it may disenfranchise contributors who cannot watch the project daily.
In such cases, a full calendar week is a fair minimum window for decision making, balancing the needs to give people enough time and to keep the process moving along.
```

All -1s votes require reason/ explanation, -1’s with no substantiation may be overturned.
Thus a single -1 vote can block achieving consensus, so it forces the group to take care of the blocking person rather than outvoting.

SIGs decide by consensus within themselves whatever they are doing, and which model is used for a particular decisions.

Anything needing wider discussion and approval goes to mailing list for either group or SIG roll-up consensus.

> When we are ready, we use this power to form a technical and/or overall steering committee, with corresponding election system.

Reviewer 2: 👍 on the consensus model

1. A check is made for any dissent on the proposal.
   1. Reservations? State reservation, sometimes with a ‘-1’ signifier
      1. Reservations about the proposal are worked through, seeking consensus to resolve the reservations.
      2. A reservation is not a vote against the proposal, but may turn into a vote against if unresolved.
      It is often expressed with an initial -1 vote to indicate reservations and concerns.
      This indicates there is still discussion to be had.
   1. Stand aside?
      No comment, or state concerns without a -1 reservation; sometimes the ‘-0’ signifier is used.
      1. This option allows a member to have issues with the proposal without choosing to block the proposal, by instead standing aside with a +/-0 vote.
      1. The stated concerns may influence other people to have or release reservations.
   1. Block?
   Vote ‘-1’ with reasons for the block.
     1. This is a complete block on a proposal, refusing to let it pass.
     A block is a -1 vote and must be accompanied with substantive arguments that are rooted in the merit criteria of the Project – protecting the community, the upstream, technical reasons, and so forth.

> ### Use of Architectural Decision Records (ADR)

Reviewer 2: not sure if we need this in the governance doc? It's important, but it's also just a technical implementation of tracking decisions.

Reviewer 12: Maybe not in the governance doc, but (as link/doc) good as a guideline/hands on.

Author: I'd like to continue this discussion a bit without blocking the Governance 1.0 acceptance. I think it's OK to go ahead with a "maybe this doesn't fit" for the Gov 1.0 release, and we can change or remove for Gov 1.1 if needed.

There are a few reasons I see ADRs as fitting in the governance:

1. Governance documents are largely about how decisions are made. _If_ an ADR is only about a specific technical implementation detail, then it is probably enough to say, "SIGs capture technical decisions in ADRs, here's a link about that. A SIG may adopt additional processes around recording decisions." That is, the governance basically leaves the details of the ADRs up to the SIGs. Perhaps one SIG ends up owning the ADR process entirely, and all of the governance around that would be within that SIG.
2. This may be my own fancy, but I've been considering what it would look like to adopt ADRs for aspects of a community architecture. That is, the possibility that an Operate First environment wants to capture decisions about how and why a specific community is architected in support of a project's mission.
3. Since an inherent effect of realizing the Operate First concept as an actual running cloud environment is creating all the code and content needed to create a functional operate first environment, it may be important to specify this kind of granular detail in the governance or the Operate First model overall in some way. That is, more so than other open source projects, it is crucial for Op1st (and all Operate First environments by extension) to have a clear and fair way to track decisions. To not do so is to fail at a core technique of operating first within the open source ecosystem.

Reviewer 12: +1 to the "without blocking" | it is OK to go ahead with a "maybe"
still need to think about @quaid 3 reasons, but they make sense to me so far

Author: I'm resolving this but we will definitely come back to this PR and gather all the open points for seeding the Gov 1.1 discussions.

ADR page: https://www.operate-first.cloud/blueprints/blueprint/

ADR repository: https://github.com/operate-first/blueprint

### Use of the Community Handbook
The Community Handbook is intended to serve as a resource for all Operate First community contributors.
It contains documentation on how to perform contributor-related tasks and encourages input and contributions from the community.
The covenant of the Community Handbook is that if contributors must figure out a task for themselves, they should submit that solution to the handbook for the benefit of future contributors.

Content for the Community Handbook is drawn from all across the project, allowing documentation to live close to the place it's used.
There is a core repository for structure, guidelines, and all content that does not fit or belong in another repository.

Community Handbook page: https://www.operate-first.cloud/community-handbook/

Community Handbook repository: https://github.com/operate-first/community-handbook

## Governance roadmap: 1.1 and beyond

This section contains content being worked on for the next iteration, Governance 1.1.

### Addition of committees, subprojects, and working Groups

> TBD

Reviewer 6: May I suggest noting which SIG or body has responsibility to decide these sections?

Author: That is an _excellent_ suggestion, as I had intended that to be clear in the governance and this is a great place to make that clear.
```suggestion
It is the purview of the initiating Community SIG to manage the Governance 1.1 process, including the responsibility to decide which types of groups get added, how they are created and ended, and where future responsibility for these type of decisions shifts once Governance 1.1 is ratified.
```

### Code of Conduct

We are using the [upstream OpenInfra Foundation Code of Conduct](https://openinfra.dev/legal/code-of-conduct)(CoC).
In creating the next Governance 1.1 we will have an actual process to make decisions as a community. We can then either formally ratify this same CoC, or the Community SIG can convene a cross-project working group to discuss and decide an inclusive and equitable process for a new code of conduct.

### Community membership
> :notebook: **Note** This document is a work in progress, especially these membership guidelines

We are opening the Governance 1.0 directly borrowing from the [Kubernetes community group definitions](https://github.com/kubernetes/community/blob/master/community-membership.md), then reviewing and refining for the Governance 1.1.

This section outlines the various responsibilities of contributor roles in Op1st.
The Op1st project is subdivided into subprojects under SIGs.
Responsibilities for most roles are scoped to these subprojects.

> A contribution can take many, many forms, such as: documentation, code, best-practices content, marketing materials, running a meeting, helping a user in chat, operating a cluster, responding to a `help wanted` flag, reviewing and commenting on contributions of all these types when in a proposed-state (a _pull request_ or _PR_), giving a talk at a conference, staffing a booth at a conference, and so forth.

Reviewer 13: ```suggestion
A contribution can take many forms, such as: documentation, code, best-practices content, marketing materials, running a meeting, helping a user in chat, operating a cluster, responding to a `help wanted` flag, reviewing and commenting on contributions of all these types when in a proposed-state (a _pull request_ or _PR_), giving a talk at a conference, staffing a booth at a conference, and so forth.
```

For a longer list, refer to this chapter, "[What Is A Contribution?](https://www.theopensourceway.org/the_open_source_way-guidebook-2.0.html#_what_is_a_contribution)", which includes these definitions:

> An open source **Contribution**
>
> Any original, intentional, and substantive object given freely to an open source community, under the licensing of that community. A contribution can come from an individual or a community.

> An open source **Contributor**
>
> Any individual person involved in making Contributions to the community. Communities are interpersonal by their nature. They consist of humans, not organizations. Organizations can send their members, staff, leaders, and so forth out to make contributions to a community as a contributor

One reason this project runs so many processes through a git workflow is to raise the visibility on all types of contributions and contributor roles.
These interactions by a contributor help qualify and quantify their activities.
It is never a full picture of one's activities, but it is a useful and helpful one.

> | Role | Responsibilities | Requirements | Defined by |

Reviewer 13: Are these roles defined at the sub-project level or are they Op1st project wide? Are they mutually exclusive, or can I be a member and an approver? Or are they hierarchical, for example, Does the the subproject owner have the same permissions as the approvers and reviewers?   

I think a little more clarity on how these roles function could be helpful here. :)

Author: I agree we need to more deeply consider this matrix in comparison with the actual way things are currently laid out and the way we'd like them to evolve. These are intended to be project-wide definitions, as written and sourced (aiui), but that is a difference we can discuss for how we want to approach it.

But I'm not sure that work is a blocker to using this as a starting place, clarifying and fixing in Gov 1.1.

What would be a minimal viable section here?

We can add a sentence about the definitions intended to be project-wide and forthcoming work on the governance will bring more clarity to this section.

If these can be edited to line up better with how we already do project roles, what would those edits look like?

Reviewer 13: Sure, this can be held until Gov 1.1. 

However, just for clarity, currently the use of  OWNERS files is per sub-project (git repo), I'm not sure if there is such a thing as a an org level OWNERS file to define these roles currently. At the org level we have a [github-config](https://github.com/operate-first/common/blob/main/github-config.yaml) where org admins, members and teams can be defined. 

It might make sense to have a note at the beginning like, "Below are the initial roles defined by Governance 1.0, where "Members" are defined across all of Op1st projects, and "Reviews", "Approvers", and "Subproject Owners" are defined per subproject. We will continue to evaluate and refine these role definitions as we move to Governance 1.1"(?)

Author: Resolving this to make clear nothing to do here for this PR, but we'll use these resolved conversations to seed the Gov 1.1. discussions.
